diff --git a/host_state_manager.cpp b/host_state_manager.cpp
index 7d661dd..a8f1d5e 100644
--- a/host_state_manager.cpp
+++ b/host_state_manager.cpp
@@ -46,6 +46,7 @@ constexpr auto HOST_STATE_POWERON_TGT = "obmc-host-start@0.target";
 constexpr auto HOST_STATE_POWERON_MIN_TGT = "obmc-host-startmin@0.target";
 constexpr auto HOST_STATE_REBOOT_TGT = "obmc-host-reboot@0.target";
 constexpr auto HOST_STATE_QUIESCE_TGT = "obmc-host-quiesce@0.target";
+constexpr auto HOST_STATE_RESET_TGT = "obmc-chassis-reset@0.target";
 
 constexpr auto ACTIVE_STATE = "active";
 constexpr auto ACTIVATING_STATE = "activating";
@@ -54,6 +55,7 @@ constexpr auto ACTIVATING_STATE = "activating";
 const std::map<server::Host::Transition, std::string> SYSTEMD_TARGET_TABLE = {
     {server::Host::Transition::Off, HOST_STATE_SOFT_POWEROFF_TGT},
     {server::Host::Transition::On, HOST_STATE_POWERON_TGT},
+    {server::Host::Transition::Reset, HOST_STATE_RESET_TGT},
     {server::Host::Transition::Reboot, HOST_STATE_REBOOT_TGT}};
 
 constexpr auto SYSTEMD_SERVICE = "org.freedesktop.systemd1";
-- 
1.8.3.1

